signing/signer.go

```diff
@@ -154,23 +163,24 @@ func Verify(pubkey *PublicKey, message []byte, sign []byte) bool {
 	return ed25519.Verify2(pubkey.Bytes(), message, sign)
 }
 
-// EDVerifier is a verifier for ED purposes.
-type EDVerifier struct{}
-
-// NewEDVerifier returns a new EDVerifier.
-func NewEDVerifier() EDVerifier {
-	return EDVerifier{}
+// NewEdVerifier returns a new EdVerifier.
+func NewEdVerifier(genesisID [20]byte) *EdVerifier {
+	return &EdVerifier{genesisID: genesisID}
 }
 
-var _ VerifyExtractor = EDVerifier{}
-
 // Verify that signature matches public key.
-func (EDVerifier) Verify(pub *PublicKey, msg, sig []byte) bool {
+func (ev EdVerifier) Verify(pub *PublicKey, msg, sig []byte) bool {
+	if len(msg) < 20 {
+		return false
+	}
+	if string(msg[0:20][:]) != string(ev.genesisID[:]) {
+		return false
+	}
```

Comment:
this implementation is incorrect. message should not include genesis id. verifier should prefix message before attempting to extract public key

Reply:
done this way in https://github.com/spacemeshos/go-spacemesh/pull/3597